File: src/main/java/pe/codember/app/Challenge01.java
package pe.codember.app;

import java.util.Arrays;
import java.util.HashMap;
import java.util.LinkedHashMap;

public class Challenge01 {

  public static String resolve(String words) {
    var wordsToArray = Arrays.asList(words.split(" "));
    var wordsCount = new LinkedHashMap<String, Integer>();

    wordsToArray.forEach(word ->
      wordsCount.put(word, (wordsCount.get(word) == null ? 0 : wordsCount.get(word)) + 1)
    );

    final StringBuilder wordConcat = new StringBuilder();

    wordsCount.forEach((word, count) -> {
      wordConcat.append(word).append(count);
    });

    return wordConcat.toString();
  }
}

File: src/main/typescript/challenge01.ts
export const countWords = (words: string): string => {
  const wordsArray = words.split(' ')
  const wordCounts = new Map<string, number>()
  wordsArray.forEach(word => {
    wordCounts.set(word, (wordCounts.get(word) ?? 0) + 1)
  })

  let wordConcat = ''

  wordCounts.forEach((count, word) => {
    wordConcat += word + count
  })

  return wordConcat
}
